wl/utils/Utils.java

```diff
@@ -462,8 +457,102 @@ public Value(String k, String v, String t) {
 		}
 	}
 
-	private static final TreeMap<ChecksummedFile, TreeMap<String, Value>> _staticprofiles =
-	    new TreeMap<>();
+	/** Wrapper around the contents of an ini-style file. */
+	public static class Profile {
+
+		/** The name of the global section in an add-ons main file. */
+		public static final String kGlobalSection = "[global]";
+
+		/** A section in a Profile. */
+		public static class Section {
+
+			/** The name of this section (may be "" for files without section divisions). */
+			public final String name;
+
+			/** The actual contents of the file. */
+			public final Map<String, Value> contents;
+
+			/**
+			 * Constructor.
+			 * @param n Name of the section.
+			 */
+			public Section(String n) {
+				name = n;
+				contents = new TreeMap<>();
+			}
+		}
+
+		private final Map<String, Section> sections;
+
+		/** Default constructor. */
+		public Profile() { sections = new TreeMap<>(); }
+
+		/**
+		 * Add a new section to the profile.
+		 * @param s The section to add.
+		 */
+		public void add(Section s) { sections.put(s.name, s); }
+
+		/**
+		 * Dump this profile to a file.
+		 * @param out Stream to print to.
+		 */
+		public void write(PrintStream out) {
+			for (Section s : sections.values()) {
```

Comment:
Now, I also did some testing and the profiles usually contain a comment about the creation
```suggestion
			out.print("# Automatically created by the Widelands Add-On Server");
			for (Section s : sections.values()) {
```